.github/workflows/test-stage.yml

```diff
@@ -8,58 +8,50 @@ on:
     types: [opened, reopened]
 
 jobs:
-  set-up-dependencies:
-    name: "Set up dependencies"
+  unit-testing:
+    name: "Unit tests"
     runs-on: ubuntu-latest
+    timeout-minutes: 5
```

Comment:
If our unit tests are taking 5 minutes something is very very wrong. I doubt we need more than the runner's default value here.